### PR TITLE
Upgrade aro cluster from current ocp version then  test with  e2e

### PR DESCRIPTION
**Pull request by mgahagan73**
Test job to validate we can upgrade an aro cluster we install in Prow then upgrade it to the latest 4.12 nightly then run e2e on it.

### CODE REVIEW

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-aro-rehearse-upgrade-e2e

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-mpg-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse openshift-tests-private-release-4.10__amd64-nightly-4.10-upgrade-from-eus-4.8

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-mpg-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.10-amd64-nightly-4.10-upgrade-from-eus-4.8-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-stable-4.10-azure-aro-p2-f7

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-aro--p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-4.13-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
ok I did not read the report script as carefully as I should have, exits with 1 because there are some e2e test failures.

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installer-rehearse-4.12-aro-rehearse-upgrade-e2e

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-aro-aro-p1-f30

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    tag: dev-scripts
+  tools:
+    name: "4.12"
+    namespace: ocp
+    tag: tools
+  upi-installer:
+    name: "4.10"
+    namespace: ocp
+    tag: upi-installer
+releases:
```
Comment (jianlinliu) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml:22]: From the following item, the upgrade path would be like 4.11 -> 4.10 -> 4.11, looks like something wrong.

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh [hunk 2]
```diff
@@ -188,12 +188,13 @@ function run_command_oc() {
...
 function check_clusteroperators() {
-    local tmp_ret=0 tmp_clusteroperator input column last_column_name tmp_clusteroperator_1 rc null_version unavailable_operator degraded_operator
+    local tmp_ret=0 tmp_clusteroperator input column last_column_name tmp_clusteroperator_1 rc null_version unavailable_operator degraded_operator skip_operator
 
+    skip_operator="aro" # ARO operator versioned but based on RP git commit ID not cluster version
     echo "Make sure every operator do not report empty column"
     tmp_clusteroperator=$(mktemp /tmp/health_check-script.XXXXXX)
     input="${tmp_clusteroperator}"
-    ${OC} get clusteroperator >"${tmp_clusteroperator}"
+    ${OC} get clusteroperator $>"${tmp_clusteroperator}"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:197]: Do you mean "&" here, but not "$", right?

Comment (mgahagan73) [reply ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh]: that was a typo, reverted to what is in master branch in latest commit.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml (hunk 1 above)
Comment (mgahagan73) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml:22]: I've changed base images to 4.11, I think it was upgrading to latest stable 4.10 -> latest stable 4.11 -> latest nightly 4.11  based on line #57 . I'll run the rehearse job again.

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.11-amd64-nightly-4.11-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.12-amd64-nightly-4.12-upgrade-from-aro-aro-p1-f30

**Comment by mgahagan73** [comment]
I think this is ready for another review now. Please let me know if there are issues with the config files, if so I could create a new PR with just the changes to the ARO provisoner and the upgrade-toimage.sh script which I think would help other team members which are depending on those fixes to make progress.

**Comment by jhou1** [comment]
Thank you @mgahagan73 , we also need to squash the commits.

/assign @jianlinliu

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml (hunk 1 above)
Comment (jianlinliu) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml:22]: I referred openshift-openshift-tests-private-release-4.12__amd64-nightly-4.12-upgrade-from-eus-4.10.yaml, per your requirement, seem like need to define `initial` and `latest` to stable 4.10, `intermediate` to stable 4.11, and `target` to nightly 4.11.

File: ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml [hunk 3]
```diff
@@ -7003,12 +7085,6 @@ periodics:
     job-release: "4.13"
     pj-rehearse.openshift.io/can-be-rehearsed: "true"
   name: periodic-ci-openshift-openshift-tests-private-release-4.13-amd64-nightly-aws-ipi-disconnected-private-p3-f28-long-duration-cloud
-  reporter_config:
```
Comment (jianlinliu) [inline ci-operator/jobs/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.13-periodics.yaml:7006]: The PR is removing some irrelevant config settings.

File: ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh [hunk 4]
```diff
@@ -188,8 +188,9 @@ function run_command_oc() {
 }
 
 function check_clusteroperators() {
-    local tmp_ret=0 tmp_clusteroperator input column last_column_name tmp_clusteroperator_1 rc null_version unavailable_operator degraded_operator
+    local tmp_ret=0 tmp_clusteroperator input column last_column_name tmp_clusteroperator_1 rc null_version unavailable_operator degraded_operator skip_operator
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/toimage/cucushift-upgrade-toimage-commands.sh:191]: cc @shellyyang1989 for awareness.

**Comment by mgahagan73** [comment]
PR has been rebased again, I attempted to squash commits and got tons of conflicts so I'm not sure if it worked correctly. If needed I'll just create a new PR with only my changes since it seems to have gotten mixed with the numerous rebases done over the past couple of months.

**Comment by jianlinliu** [comment]
> PR has been rebased again, I attempted to squash commits and got tons of conflicts so I'm not sure if it worked correctly. If needed I'll just create a new PR with only my changes since it seems to have gotten mixed with the numerous rebases done over the past couple of months.

I suggest to create a new PR.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml (hunk 1 above)
Comment (shellyyang1989) [reply ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml:22]: If we use 
```
commands: echo "${RELEASE_IMAGE_INTERMEDIATE410},${RELEASE_IMAGE_STABLE},${RELEASE_IMAGE_TARGET}"
        > ${SHARED_DIR}/upgrade-edge
```
the upgrade path is gonna be RELEASE_IMAGE_LATEST -> RELEASE_IMAGE_INTERMEDIATE410 -> RELEASE_IMAGE_STABLE -> RELEASE_IMAGE_TARGET.

What's your expected upgrade path?

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+        name: release:intermediate410
+      - env: RELEASE_IMAGE_TARGET
+        name: release:target
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - ref: cucushift-upgrade-toimage
+    - chain: openshift-e2e-test-qe
```
Comment (shellyyang1989) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml:72]: We can use `- chain: openshift-upgrade-qe-test` instead which includes pre-upgrade test and post-upgrade test.

File: ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+resources:
+  '*':
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: aro-p1-f30
+  cron: 59 13 20 * *
+  steps:
+    cluster_profile: azure-qe
```
Comment (shellyyang1989) [inline ci-operator/config/openshift/openshift-tests-private/openshift-openshift-tests-private-release-4.11__amd64-nightly-4.11-upgrade-from-aro.yaml:54]: We may need some env set:

```
env:
      BASE_DOMAIN: qe.azure.devcluster.openshift.com
      UPGRADE_CHECK_RUN_TAGS: '@upgrade-check and @4.10 and @azure-ipi and not @fips'
      UPGRADE_PRE_RUN_TAGS: '@upgrade-prepare and @4.10 and @azure-ipi and not @fips'
```

**Pull request closed without merging**